### PR TITLE
Implement some granular logic to --cache

### PR DESCRIPTION
**Pull request by DevonianTeuchter**
Allow multiple package specifications for --unused and --cache
Allow --all  for --unused and --cache  [--cache with no parameters defaults to --all]
Allow both --unused and --cache to be specified together (with a package ilst or --all)

### CODE REVIEW

**Comment by DevonianTeuchter** [comment]
Initial PR tests expected to fail with new parameters for clean.  Additional testing added via metaport PR

File: bin/zopen-clean [hunk 1]
```diff
@@ -102,7 +108,7 @@ cleanDangling()
   killph="kill -HUP ${ph}"
   addCleanupTrapCmd "${killph}"
   zosfind "${ZOPEN_ROOTFS}" -type l -exec test ! -e {} \; -print | while read sl; do
-    printVerbose "Removing symlink '${sl}'"
+    printVerbose "Removing dahngling symlink '${sl}'"
```
Comment (IgorTodorovskiIBM) [inline bin/zopen-clean:111]:
```suggestion
    printVerbose "Removing dangling symlink '${sl}'"
```

File: bin/zopen-clean [hunk 2]
```diff
@@ -50,28 +52,32 @@ HELPDOC
...
+    cd "${ZOPEN_PKGINSTALL}/${needle}" && zosfind . -name "./*" -prune -type d |
+    while read repo; do
+      printVerbose "Parsing repo: '${repo}' as '${repo#./}'"
+      repo="${repo#./}"
+      if [ "${current}" = "${repo}" ]; then
+        printInfo "${NC}${GREEN}-> ${repo}  <- Current version${NC}"
+      else
+        deref=$(cd "${ZOPEN_PKGINSTALL}/${needle}/${repo}" && pwd -P)
+        [ "${deref}" = "/" ] || rm -rf "${deref}" > /dev/null 2>&1
+        printInfo "-- ${repo} <- Removed"
```
Comment (IgorTodorovskiIBM) [inline bin/zopen-clean:76]: Should we add a prompt before removal?

Also, I wonder if any of the removal logic can be shared with zopen-remove and placed into a function in common.sh?

File: bin/zopen-clean [hunk 3]
```diff
@@ -111,10 +117,21 @@ cleanDangling()
...
+    printVerbose "Cleaning all entries in ${ZOPEN_ROOTFS}/var/cache/zopen"
+    rm -rf "${ZOPEN_ROOTFS}/var/cache/zopen/"*
+    syslog "${ZOPEN_LOG_PATH}/audit.log" "${LOG_A}" "${CAT_FILE}" "CLEAN" "cleanCache" "Main cache in ${ZOPEN_ROOTFS}/var/cache/zopen cleaned"
+    printInfo "- Cache at '${ZOPEN_ROOTFS}/var/cache/zopen' cleaned"
+  else
+    for needle in $1; do
+      printVerbose "Cleaning ${ZOPEN_ROOTFS}/var/cache/zopen entries for '${needle}"
+      rm -rf "${ZOPEN_ROOTFS}"/var/cache/zopen/"${needle}"-*
+      syslog "${ZOPEN_LOG_PATH}/audit.log" "${LOG_A}" "${CAT_FILE}" "CLEAN" "cleanCache" "Cache for '${needle}' in ${ZOPEN_ROOTFS}/var/cache/zopen cleaned."
+      printVerbose "- Cleaned cached files for '${needle}' in cache at '${ZOPEN_ROOTFS}/var/cache/zopen'."
```
Comment (IgorTodorovskiIBM) [inline bin/zopen-clean:132]: Should we add an option to clean the json cache? 
zopen_releases.json
zopen_releases.timestamp
zopen_releases.timestamp.current

**Comment by DevonianTeuchter** [comment]
@IgorTodorovskiIBM  Implemented a --metadata cleanup switch.  Note tests will fail again until corresponding PR in metaport is merged

**Review by IgorTodorovskiIBM — APPROVED** [review]
Great! LGTM

**Comment by DevonianTeuchter** [comment]
Tests now passing after metaport PR merged

**Pull request merged**